tools/clang/test/DXC/Inputs/bom-main-ascii.hlsl

```diff
@@ -0,0 +1,20 @@
+// Tests main and include files with/without BOM to ensure BOM is stripped.
+
+#include "bom-inc-ascii.hlsli"
+#include "bom-inc-utf8.hlsli"
+#include "bom-inc-utf16le.hlsli"
+
+// TODO: Add support for Big Endian and UTF-32
```

Comment:
Do we have a GitHub issue for this?

Reply:
I don't think we have anything tracking this.  There probably should be two separate issues.

1. Test BOM on Linux with utf32le in place of utf16le.
2. Expand the compiler's Unicode support to more formats, such as big endian formats, utf-32 on Win32, and utf-16 on Linux/Mac.

Reply:
Added https://github.com/microsoft/DirectXShaderCompiler/issues/4932 and https://github.com/microsoft/DirectXShaderCompiler/issues/4931